src/nvim/ops.c

```diff
@@ -2524,6 +2528,59 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+/// Execute autocommands for TextYankPost.
+///
+/// @param oap Operator arguments.
+/// @param reg The yank register used.
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+  FUNC_ATTR_NONNULL_ALL
+{
+  static bool recursive = false;
+
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST) || recursive) {
```

Comment:
might as well check `recursive` first, and declare `buf` nearer to its first use.